### PR TITLE
Hide out of stock for configurable products (move PR #378)

### PR DESCRIPTION
**Pull request by vmalyk**
**Summary**

After 1.12.0 and PR #841  Issue is back. Configurable product is set "In stock" but all its simple variations have qty 0, this product in fact shows as not available on frontend and shouldn't go to Algolia.
 
**Result**

Applied fix from #378 to new method productIsInStock.
PR is created for 1.x branches, but Fix is actual for 2.0 too.

### CODE REVIEW

**Comment by vmalyk** [comment]
@damcou , please review,

thank you!

File: Helper/Entity/ProductHelper.php [hunk 1]
```diff
@@ -1163,10 +1163,15 @@ public function canProductBeReindexed($product, $storeId, $isChildProduct = fals
...
+     * @param Product $product
+     * @param int $storeId
+     * @return bool|int
+     */
     public function productIsInStock($product, $storeId)
     {
         $stockItem = $this->stockRegistry->getStockItem($product->getId());
 
-        return $stockItem->getIsInStock();
+        return ($product->isSaleable() && $stockItem->getIsInStock());
```
Comment (damcou) [inline Helper/Entity/ProductHelper.php]:
```suggestion
        return $product->isSaleable() && $stockItem->getIsInStock();
```

Comment (vmalyk) [reply Helper/Entity/ProductHelper.php]: @damcou , thank you for review!

Brackets have been removed.
Branch develop-1.x has been merged.

File: Helper/Entity/ProductHelper.php [hunk 2]
```diff
@@ -1163,10 +1163,15 @@ public function canProductBeReindexed($product, $storeId, $isChildProduct = fals
         return true;
     }
 
+    /**
+     * @param Product $product
+     * @param int $storeId
```
Comment (damcou) [inline Helper/Entity/ProductHelper.php]:
```suggestion
     * @param int $storeId
     
```
Sorry, I just checked the code '-_-

File: Helper/Entity/ProductHelper.php (hunk 1 above)
Comment (damcou) [reply Helper/Entity/ProductHelper.php]: Sorry, I missed the PHPdoc issue ;)

File: Helper/Entity/ProductHelper.php (hunk 2 above)
Comment (vmalyk) [reply Helper/Entity/ProductHelper.php]: @damcou, Fixed PHPdoc and added extra line between params and return, added method description.

**Pull request merged**